File: Lista/Go/Q6.go
package main

import "fmt"

type Motor struct {
    Potencia int
    Tipo     string
}

func (m Motor) ExibirDetalhes() {
    fmt.Printf("Motor: %s com %d cavalos de potência\n", m.Tipo, m.Potencia)
}

type Carro struct {
    Marca  string
    Modelo string
    Ano    int
    Motor  Motor
}

func (c Carro) ExibirDetalhes() {
    fmt.Printf("Carro: %s %s, Ano: %d\n", c.Marca, c.Modelo, c.Ano)
    c.Motor.ExibirDetalhes()
}

func main() {
    motor1 := Motor{
        Potencia: 180,
        Tipo:     "V8",
    }

    carro1 := Carro{
        Marca:  "Ford",
        Modelo: "Mustang",
        Ano:    2022,
        Motor:  motor1,
    }

    carro1.ExibirDetalhes()
}

File: Lista/Java/Q6.java
class Motor {
    private String tipo;
    private int potencia;

    public Motor(String tipo, int potencia) {
        this.tipo = tipo;
        this.potencia = potencia;
    }

    public void exibirInformacoes() {
        System.out.println("Tipo do motor: " + tipo + ", Potência: " + potencia + " HP");
    }
}

class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private Motor motor;

    public Carro(String marca, String modelo, int ano, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.motor = motor;
    }

    public void exibirDetalhes() {
        System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano);
        motor.exibirInformacoes();
    }
}

public class Main {
    public static void main(String[] args) {
        Motor motor1 = new Motor("Gasolina", 150);

        Carro carro1 = new Carro("Ford", "Mustang", 2022, motor1);

        carro1.exibirDetalhes();
    }
}
